cmd/common_test.go

```diff
@@ -1,71 +1,201 @@
+	}
 }
 
-func (s *CommonTestSuite) TestParseClusterAndProjectID(c *check.C) {
-	testParse(c, "local:p-12345", "local", "p-12345", false)
-	testParse(c, "c-12345:p-12345", "c-12345", "p-12345", false)
-	testParse(c, "cocal:p-12345", "", "", true)
-	testParse(c, "c-123:p-123", "", "", true)
-	testParse(c, "", "", "", true)
-	testParse(c, "c-m-12345678:p-12345", "c-m-12345678", "p-12345", false)
-	testParse(c, "c-m-123:p-12345", "", "", true)
-}
+func TestConvertSnakeCaseKeysToCamelCase(t *testing.T) {
+	t.Parallel()
 
-func (s *CommonTestSuite) TestConvertSnakeCaseKeysToCamelCase(c *check.C) {
-	cases := []struct {
-		input   map[string]interface{}
-		renamed map[string]interface{}
+	tests := []struct {
+		input map[string]any
+		want  map[string]any
 	}{
 		{
-			map[string]interface{}{"foo_bar": "hello"},
-			map[string]interface{}{"fooBar": "hello"},
+			map[string]any{"foo_bar": "hello"},
+			map[string]any{"fooBar": "hello"},
+		},
+		{
+			map[string]any{"fooBar": "hello"},
+			map[string]any{"fooBar": "hello"},
+		},
+		{
+			map[string]any{"foobar": "hello", "some_key": "valueUnmodified", "bar-baz": "bar-baz"},
+			map[string]any{"foobar": "hello", "someKey": "valueUnmodified", "bar-baz": "bar-baz"},
 		},
 		{
-			map[string]interface{}{"fooBar": "hello"},
-			map[string]interface{}{"fooBar": "hello"},
+			map[string]any{"foo_bar": "hello", "backup_config": map[string]any{"hello_world": true}, "config_id": 123},
+			map[string]any{"fooBar": "hello", "backupConfig": map[string]any{"helloWorld": true}, "configId": 123},
 		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run("", func(t *testing.T) {
+			t.Parallel()
+
+			convertSnakeCaseKeysToCamelCase(test.input)
+			assert.Equal(t, test.input, test.want)
+		})
+	}
+}
+
+func TestParsePrincipalID(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		id   string
+		want *managementClient.Principal
+	}{
 		{
-			map[string]interface{}{"foobar": "hello", "some_key": "valueUnmodified", "bar-baz": "bar-baz"},
-			map[string]interface{}{"foobar": "hello", "someKey": "valueUnmodified", "bar-baz": "bar-baz"},
+			id: "local://user-2p7w6",
+			want: &managementClient.Principal{
+				Name:          "user-2p7w6",
+				LoginName:     "user-2p7w6",
+				Provider:      "local",
+				PrincipalType: "user",
+			},
 		},
 		{
-			map[string]interface{}{"foo_bar": "hello", "backup_config": map[string]interface{}{"hello_world": true}, "config_id": 123},
-			map[string]interface{}{"fooBar": "hello", "backupConfig": map[string]interface{}{"helloWorld": true}, "configId": 123},
+			id: "okta_group://b4qkhsnliz",
+			want: &managementClient.Principal{
+				Name:          "b4qkhsnliz",
+				LoginName:     "b4qkhsnliz",
+				Provider:      "okta",
+				PrincipalType: "group",
+			},
 		},
 	}
 
-	for _, tc := range cases {
-		convertSnakeCaseKeysToCamelCase(tc.input)
-		c.Assert(tc.input, check.DeepEquals, tc.renamed)
+	for _, test := range tests {
+		test := test
+		t.Run("", func(t *testing.T) {
```

Comment:
How about outputting `test.id` instead of `""` ?

Reply:
Of course.